Fix getPoolStateWithBalancesV3 helper

brunoguerios opened the pull request:
Closes #561 

getPoolStateWithBalances should use raw balances instead of live balances when used to calculate proportional amounts

this PR fixes an issue where karpatkey is trying to add liquidity to the [sDAI/BRLA pool](https://balancer.fi/pools/gnosis/v3/0xa91c3c043051e7e680b61d79b3a733d3e2c0fb5e) on gnosis chain

mkflow27 reviewed (approved):
lgtm

johngrantuk reviewed (approved):
Any way we can add a test that covers the current issue?

brunoguerios commented:
I added the test but it's failing because anvil is not finding the forkUrl 🤔 
I'm assuming it's related to the test running on gnosis-chain, but this error doesn't make much sense to me.
ANVIL_NETWORKS is properly set with GNOSIS_CHAIN config, including a `fallBackRpc` url
In any case, I added a `GNOSIS_CHAIN_RPC_URL` to our repository secrets, but still didn't work
I'm testing against this specific pool, because that's the one that originally highlighted the issue
Let me know if you guys have any suggestion on how to handle this
